### PR TITLE
2022q1 fpart

### PR DESCRIPTION
**Pull request by martymac**
Hello,

Here is a suggested entry for fpart and fpsync, two tools I am working on during my spare time.

Best regards,

Ganael.

### CODE REVIEW

File: 2022q1/fpart.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Contact: Ganael Laplanche <martymac@FreeBSD.org>
+
+==== What is Fpart ?
+
+Fpart is a *Filesystem partitioner*. It helps you sort file trees and pack them into bags ("partitions").
+
+==== What is Fpsync ?
+
+Fpsync is a companion script that uses Fpart underhood to parallelize rsync(1) or cpio(1) jobs, making it a simple but powerful data migration tool. Those jobs can be run either locally or remotely (using SSH). Fpsync is link:https://www.fpart.org/links/[often used by researchers and cloud providers] where lots of data need to be moved and clusters are available to speed up transfers.
```
Comment (pauamma) [inline 2022q1/fpart.adoc:16]:
```suggestion
Fpsync is a companion script that uses Fpart under the hood to parallelize rsync(1) or cpio(1) jobs, making it a simple but powerful data migration tool. Those jobs can be run either locally or remotely (using SSH). Fpsync is link:https://www.fpart.org/links/[often used by researchers and cloud providers] where lots of data need to be moved and clusters are available to speed up transfers.
```

File: 2022q1/fpart.adoc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+==== Q1 2022 Status
+
+Both tools continued to evolve and saw several bugs fixed, see the link:https://www.fpart.org/changelog/[changelog].
```
Comment (pauamma) [inline 2022q1/fpart.adoc:20]:
```suggestion
Both tools continued to evolve and saw several bugs fixed; see the link:https://www.fpart.org/changelog/[changelog].
```

File: 2022q1/fpart.adoc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+It should be merged soon (hopefully).
+
+Last but not least, Fpart has been link:https://sill.etalab.gouv.fr/fr/software?id=229[referenced in the French Government's 'SILL'].
+
+==== Contributing
+
+If you are interested in contributing, have a look at the link:https://github.com/martymac/fpart/blob/master/TODO[TODO list].
+
+Any contribution is welcome, more especially in the field of unit testing.
```
Comment (pauamma) [inline 2022q1/fpart.adoc:34]: "more especially" (even in the "particularly" sense) looks wrong, but not enough to  request a change without a native English speaker confirming.

**Review by pauamma — CHANGES_REQUESTED** [review]

File: 2022q1/fpart.adoc (hunk 3 above)
Comment (martymac) [reply 2022q1/fpart.adoc:34]: Hello @pauamma,

Yes, for that change, let's wait for a native english-speaker's confirmation. Thanks for your feedback!

**Comment by martymac** [comment]
> This might be a stupid question, but what does fpart have to do with FreeBSD, specifically?

Hello Daniel,

This is not a stupid question : there is no direct relation between FreeBSD and Fpart, but Fpart is primarily developped and tested *on* FreeBSD. It also embeds its fts(3) library (as an option, which led to a patch currently being reviewed, see the report). Also, storage and networking being two key areas for FreeBSD, I thought it may be of interest to our users.

This entry is only a *suggestion*, and for the  "Third-party project" section (of course).

Cheers,

Ganael.

**Comment by debdrup** [comment]
> This is not a stupid question : there is no direct relation between FreeBSD and Fpart, but Fpart is primarily developped and tested _on_ FreeBSD. It also embeds its fts(3) library (as an option, which led to a patch currently being reviewed, see the report). Also, storage and networking being two key areas for FreeBSD, I thought it may be of interest to our users.

I completely missed the paragraph about using fts(3), but that definitely qualifies it for being part of the status report. :)

**Comment by martymac** [comment]
> Since I missed it, maybe it'd be an idea to make it more explicit that it's using fts(3) from FreeBSD, rather than it being an aside in a parenthesis?

Good idea! I've pushed an update in that way. Thanks for your feedback!

File: 2022q1/fpart.adoc [hunk 4]
```diff
@@ -7,25 +7,29 @@ link:https://www.freshports.org/sysutils/fpart[Port] URL: link:https://www.fresh
...
+==== What is fpart ?
 
-Fpart is a *Filesystem partitioner*. It helps you sort file trees and pack them into bags ("partitions").
+Fpart is a *filesystem partitioner*. It helps you sort file trees and pack them into bags ("partitions").
 
-==== What is Fpsync ?
+It uses FreeBSD's fts(3) implementation (GNU/Linux builds can also use it as an option), which makes it crawl filesystems very fast.
 
-Fpsync is a companion script that uses Fpart underhood to parallelize rsync(1) or cpio(1) jobs, making it a simple but powerful data migration tool. Those jobs can be run either locally or remotely (using SSH). Fpsync is link:https://www.fpart.org/links/[often used by researchers and cloud providers] where lots of data need to be moved and clusters are available to speed up transfers.
+A hook facility is provided to trigger actions on produced partitions.
```
Comment (pauamma) [inline 2022q1/fpart.adoc:16]:
```suggestion
A hook facility is provided to trigger actions on the partitions produced.
```

File: 2022q1/fpart.adoc [hunk 5]
```diff
@@ -7,25 +7,29 @@ link:https://www.freshports.org/sysutils/fpart[Port] URL: link:https://www.fresh
...
+
+Fpsync is a companion script that uses fpart under the hood to parallelize rsync(1) or cpio(1) jobs, making it a simple but powerful data migration tool. Those jobs can be run either locally or remotely (using SSH). Fpsync is link:https://www.fpart.org/links/[often used by researchers and cloud providers] where lots of data need to be moved and clusters are available to speed up transfers.
 
 ==== Q1 2022 Status
 
-Both tools continued to evolve and saw several bugs fixed, see the link:https://www.fpart.org/changelog/[changelog].
+Both tools continued to evolve and saw several bugs fixed; see the link:https://www.fpart.org/changelog/[changelog].
 
-Also, a user reported an interesting bug regarding FreeBSD's fts(3) implementation (embedded in Fpart as an option), which ignores readdir(3) errors. I have reported the bug in our Bugzilla:
+Also, a user reported an interesting bug regarding our fts(3) implementation, which ignores readdir(3) errors. I have reported the bug in our Bugzilla:
```
Comment (pauamma) [inline 2022q1/fpart.adoc:26]:
```suggestion
Also, a user reported a major bug regarding our fts(3) implementation, which ignores readdir(3) errors. I have reported the bug in our Bugzilla:
```
Value judgement, but "interesting" is too vague.

**Review by pauamma — APPROVED** [review]
Would approve with the suggested changes

**Pull request merged**